src/main/scala/where/Predicate.scala

```diff
@@ -1,10 +1,32 @@
 package com.originate.scalypher.where
 
 import com.originate.scalypher.ConstantString
+import com.originate.scalypher.path.Path
 import com.originate.scalypher.ToQuery
 
-sealed trait Predicate extends ToQuery
+case class PredicateWithProjection(
+  predicate: Predicate,
+  projection: PathProjection
+) {
+  def where(f: ObjectReference => Where): PredicateCondition =
```

Comment:
this `nodesIn` vs `nodeIn` aliases only exist for grammatical reasons, not for polymorphic reasons:
```scala
All nodesIn path where { ... }
Single nodeIn path where { ... }
```
you still end up iterating over a single object reference